File: core/src/main/java/heroes/journey/entities/items/ItemSubType.java
package heroes.journey.entities.items;

import heroes.journey.registries.Registrable;
import lombok.Getter;

@Getter
public class ItemSubType extends Registrable {

    private final String name;
    private final ItemType parentType;

    public ItemSubType(String id, String name, ItemType parentType) {
        super(id);
        this.name = name;
        this.parentType = parentType;
    }

    public String toString() {
        return name;
    }

}

File: core/src/main/java/heroes/journey/entities/items/ItemSubType.kt
package heroes.journey.entities.items

import heroes.journey.modlib.items.IItemSubType
import heroes.journey.modlib.items.ItemType
import heroes.journey.modlib.registries.Registrable
import heroes.journey.mods.Registries.ItemSubTypeManager
import lombok.Getter

@Getter
class ItemSubType(id: String, override val parentType: ItemType) : Registrable(id), IItemSubType {
    override fun register(): ItemSubType {
        return ItemSubTypeManager.register(this)
    }
}
